NEWS.md: clarify `--trim` (#56460)
JuliaLang/julia: NEWS.md

New language features
---------------------

> - New option `--trim` for building "trimmed" binaries, where code provably not reachable from entry points
>   is removed. Entry points can be marked using `Base.Experimental.entrypoint` ([#55047]).

Reviewer: How about a rewording like this?

```suggestion
- New option `--trim` creates smaller binaries by removing code that was not proven to be reachable
 from the entry points. Entry points can be marked using `Base.Experimental.entrypoint` ([#55047]).
```

Author: That wording seems fine but is still slightly inverted: it should be "proven not to be reachable" rather than "not proven to be reachable"

Author: Ah, I only now saw the other comments above. That's kinda scary that it is indeed inverted as described above.

New language features
---------------------

> - New option `--trim` creates smaller binaries by removing code that was proven not to be statically reachable from

Reviewer: We still need to use "not proven" over "proven not" —

```suggestion
- New option `--trim` creates smaller binaries by removing code that was not proven to be statically reachable from
```

Reviewer 2: Correct. I also think "statically" is kind of jargony and not necessary.

Author: done
